src/v/serde/serde.h

```diff
@@ -713,6 +729,20 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
         for (auto i = 0U; i < size; ++i) {
             t.push_back(read_nested<value_type>(in, bytes_left_limit));
         }
+    } else if constexpr (reflection::is_std_array<Type>) {
```

Comment:
whats the implication of this? Curious i don't know myself but would the compiler generate a new branch of generated code here once per every type of std::array with a different size parameter?

Reply:
> but would the compiler generate a new branch of generated code here once per every type of std::array with a different size parameter?

yeh, that's the way i understand it. i think that alone is a good argument for not supporting std::array. OTOH, we probably have very few instances of it...

Reply:
Maybe we should isntead support `std::span<T, std::dynamic_extent>`?

Reply:
it's probably ok unless we have any good alternative ideas. the most important thing is that we keep an on-disk format that doesn't constrain us much. and i think we have here: its the same as vector, for exmaple.

Reply:
yes std::span seems a good idea. let me try

Reply:
https://gcc.godbolt.org/z/sf3GoeoGj did a quick test. gcc will produce reasonable code, but clang needs __attribute__((noinline)). 
If it's ok i can use this code.
another option is to only add support for std::span in serde, and i can use serde_write/serde_read to convert std::array to std::span.
let me know if I should go with the first or the second option

Reply:
Yeah i'd vote to keep it as it is